PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py

```diff
@@ -818,21 +824,16 @@ def getMETUncertainties(self, process, metType, metModName, electronCollection,
             pfCandsNoJets = cms.EDProducer("CandPtrProjector", 
                                            src = pfCandCollection, 
                                            veto = copy.copy(jetCollection),
-                                           useDeltaRforFootprint = cms.bool(False)
                                            )
-            if self.getvalue("Puppi"):
-              pfCandsNoJets.useDeltaRforFootprint = True
             addToProcessAndTask("pfCandsNoJets"+postfix, pfCandsNoJets, process, task)
             metUncSequence += getattr(process, "pfCandsNoJets"+postfix)
 
             #electron projection ==
             pfCandsNoJetsNoEle = cms.EDProducer("CandPtrProjector", 
                                                 src = cms.InputTag("pfCandsNoJets"+postfix),
                                                 veto = copy.copy(electronCollection),
-                                                useDeltaRforFootprint = cms.bool(False),
                                                 )
             if not self.getvalue("onMiniAOD"):
-              pfCandsNoJetsNoEle.useDeltaRforFootprint = True
               pfCandsNoJetsNoEle.veto = "pfeGammaToCandidate:electrons"
             addToProcessAndTask("pfCandsNoJetsNoEle"+postfix, pfCandsNoJetsNoEle, process, task)
```

Comment:
I was debugging  differences in UL re-miniAOD between 106X and 112X, and was puzzled by the somewhat small but numerous changes in slimmedMETsNoHF UnclusteredEnUp/Down.
If I understood correctly (although I did not add any debug printouts to confirm), the problem is that the "NoHF" is based on ```noHFCands```, which are copies (not Ptrs) of the original PF cands https://github.com/cms-sw/cmssw/blob/3402e097caee10a4a7205dc67a80108939870f74/CommonTools/ParticleFlow/interface/GenericPFCandidateSelectorDefinition.h#L31-L34
So, the CandPtrProjector can not match the vetoing electron's sourceCandidatePtr (the original particleFlow) to the ```noHFCands``` (another check by the cand's sourceCandidatePtr may suffice though).
So, either we need an extension in CandPtrProjector to also check for the main collection sourceCandidatePtr or instead have ```useDeltaRforFootprint = True```

I'll re-post this as an issue.